src/python/WMCore/Services/DBS/DBS3Reader.py

```diff
@@ -339,7 +339,9 @@ def crossCheck(self, datasetPath, *lfns):
         Return the list of lfns that are in the dataset
 
         """
-        allLfns = self.dbs.listFileArray(dataset=datasetPath, validFileOnly=1, detail=False)
+        allLfns = []
+        for fileDict in self.dbs.listFileArray(dataset=datasetPath, validFileOnly=1, detail=False):
```

Comment:
The call to `self.dbs.listFileArray` may fail and throw exception, as such the for loop here will break on exception. I don't know if you want to do it this way or if you actually want to put this API call into `try/except` block. I will do the later and throw proper exception as you do in other parts of the code, see line 144.

Reply:
I did not want to make too many changes in here, outside of the scope of the original issue, but I think it's a fair request. I will get this code updated soon.